cmd/flat-network-controller/app/dns_deployment.go

```diff
@@ -0,0 +1,379 @@
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("unexpected error to get cluster role binding %s/%s: %v", net.Tenant, dnsRoleBindingName, err)
+		}
+
+		rolebinding := &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   dnsRoleBindingName,
+				Tenant: net.Tenant,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      dnsBaseName,
+					Namespace: metav1.NamespaceSystem,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     dnsRoleName,
+			},
+		}
+		if _, err := client.RbacV1().ClusterRoleBindingsWithMultiTenancy(net.Tenant).Create(rolebinding); err != nil {
+			if !errors.IsAlreadyExists(err) {
+				return fmt.Errorf("falied to create cluster role binding %s/%s: %v", net.Tenant, dnsRoleBindingName, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func ensureToCreateConfigMap(net *v1.Network, client kubernetes.Interface, domainName string) error {
+	name := dnsBaseName + "-" + net.Name
+	if _, err := client.CoreV1().ConfigMapsWithMultiTenancy(metav1.NamespaceSystem, net.Tenant).Get(name, metav1.GetOptions{}); err != nil {
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("unexpected error to get configmap %s/%s/%s: %v", net.Tenant, metav1.NamespaceSystem, name, err)
+		}
+
+		corefile, err := generateCorefile(net.Name, domainName)
+		if err != nil {
+			return fmt.Errorf("falied to create configmap %s/%s/%s: %v", net.Tenant, metav1.NamespaceSystem, name, err)
+		}
+
+		configmap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Tenant:    net.Tenant,
+				Namespace: metav1.NamespaceSystem,
+			},
+			Data: map[string]string{
+				"Corefile": corefile,
+			},
+		}
+		if _, err := client.CoreV1().ConfigMapsWithMultiTenancy(metav1.NamespaceSystem, net.Tenant).Create(configmap); err != nil {
+			if !errors.IsAlreadyExists(err) {
+				return fmt.Errorf("falied to create configmap %s/%s/%s: %v", net.Tenant, metav1.NamespaceSystem, name, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func ensureToCreateDeployment(net *v1.Network, client kubernetes.Interface) error {
+	name := dnsBaseName + "-" + net.Name
+	label := dnsServiceDefaultName + "-" + net.Name
+	configmap := dnsBaseName + "-" + net.Name
+	readOnlyRootFilesystem := true
+
+	if _, err := client.AppsV1().DeploymentsWithMultiTenancy(metav1.NamespaceSystem, net.Tenant).Get(name, metav1.GetOptions{}); err != nil {
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("unexpected error to get deployment %s/%s/%s: %v", net.Tenant, metav1.NamespaceSystem, name, err)
+		}
+
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Tenant:    net.Tenant,
+				Namespace: metav1.NamespaceSystem,
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"k8s-app": label,
```

Comment:
throughout the files: Replace "k8s-app" with clusterAddonLabelKey

Seems to me we also have a const for "arktos.futurewei.com/network"

Reply:
will substitute with consts as suggested.